ArcSlider/ArcSliderBehaviorDecorator.js

```diff
@@ -0,0 +1,121 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {positionToAngle} from '../Arc/utils';
+import {angleToValue} from './utils';
+
+// Adds agate-specific arcSlider behaviors
+const ArcSliderBehaviorDecorator = hoc((config, Wrapped) => {
+	return class extends React.Component {
+		static displayName = 'ArcSliderBehaviorDecorator';
+
+		static propTypes = {
+			endAngle: PropTypes.number,
+			max: PropTypes.number,
+			min: PropTypes.number,
+			radius: PropTypes.number,
+			startAngle: PropTypes.number,
+			step: PropTypes.number,
+			strokeWidth: PropTypes.number
+		};
+
+		constructor (props) {
+			super(props);
+
+			this.componentRef = React.createRef();
+
+			this.state = {
+				value: props.min
+			};
+		}
+
+		onMouseDown = (ev) => {
+			const componentRef = this.componentRef.current;
+
+			if (!platform.touchscreen) {
+				if (componentRef) {
+					componentRef.addEventListener('mousemove', this.calculateNewValue);
+					componentRef.addEventListener('mouseup', this.removeMouseListeners);
+				}
+				this.calculateNewValue(ev);
+			}
+		};
+
+		onTouchStart = (ev) => {
+			const componentRef = this.componentRef.current;
+
+			if (componentRef) {
+				componentRef.addEventListener('touchmove', this.calculateNewValue);
+				componentRef.addEventListener('touchend', this.removeMouseListeners);
+			}
+			this.calculateNewValue(ev);
+
+		};
+
+		removeMouseListeners = () => {
+			const componentRef = this.componentRef.current;
+			if (componentRef) {
+				componentRef.removeEventListener('mousemove', this.calculateNewValue);
+				componentRef.removeEventListener('touchmove', this.calculateNewValue);
+				componentRef.removeEventListener('mouseup', this.removeMouseListeners);
+				componentRef.removeEventListener('touchend', this.removeMouseListeners);
+			}
+		};
+
+		// Calculates the new SVG value based on the mouse cursor coordinates and sets the new value into the state
+		calculateNewValue = (ev) => {
+			const {endAngle, max, min, radius, startAngle, step, strokeWidth} = this.props;
+
+			const componentRef = this.componentRef.current;
+			if (!componentRef) {
+				return;
+			}
+			// Find the coordinates with respect to the SVG
+			const svgPoint = componentRef.createSVGPoint();
+			if (platform.touchscreen) {
+				svgPoint.x = ev.touches[0].clientX;
+				svgPoint.y = ev.touches[0].clientY;
+			} else {
+				svgPoint.x = ev.clientX;
+				svgPoint.y = ev.clientY;
+			}
+			const coordsInSvg = svgPoint.matrixTransform(componentRef.getScreenCTM().inverse());
+
+			const angle = positionToAngle(coordsInSvg, radius * 2 - strokeWidth);
+
+			// get the value based on the angle, min and max
+			let value = angleToValue(angle, min, max, startAngle, endAngle);
+
+			// adjust value based on the step
+			if (step) {
+				const delta = (value - min) % step;
+				if (delta < step / 2) {
+					value -= delta;
+				} else {
+					value += step - delta;
+				}
+			}
+
+			this.setState({value: value});
```

Comment:
We need to forward `onChange` event when changing a value so that the App could know it.